hscommon/hsutil/imagebuttonhack.go

```diff
@@ -2,23 +2,19 @@ package hsutil
 
 import (
 	"github.com/ianling/giu"
-	"github.com/ianling/imgui-go"
 )
 
-// MakeImageButton is a hack for giu.ImageButton that manually sets the ImageButton ID by
-// manually setting the ID through imgui
-//nolint:unparam // width and height are always 15 at the time of writing, but may change
+// it looks like childWidget's size must be a bit greater then widget iside of it
+const (
+	childSizeModifier = 8
+)
+
+// MakeImageButton is a hack for giu.ImageButton that creates image button
+// as a giu.child
 func MakeImageButton(id string, w, h int, t *giu.Texture, fn func()) giu.Layout {
 	return giu.Layout{
-		giu.ImageButton(t).Size(float32(w), float32(h)).OnClick(fn),
-		giu.Custom(func() {
-			// make this button unique across all editor instances
-			// at the time of writing, ImageButton uses the texture ID as the button ID
-			// so it wont be unique across multiple instances if we use the same texture...
-			// we need to step over giu and manually tell imgui to pop the last ID and
-			// push the desired one onto the stack
-			imgui.PopID()
-			imgui.PushID(id)
-		}),
+		giu.Child(id+"child").Border(false).Size(float32(w+childSizeModifier), float32(h+childSizeModifier)).Layout(giu.Layout{
+			giu.ImageButton(t).Size(float32(w), float32(h)).OnClick(fn),
+		}).Flags(giu.WindowFlagsNoDecoration),
 	}
```

Comment:
just my opinion, but i find what you have here hard to read. There's no reason why we can't just have a few more lines of code and comment what we're doing _so that other people can read it_.
```golang
	const (
		childIdSuffix = "child"
		padOneSide    = 4               // pixels
		padBothSides  = padOneSide << 1 // pixels
	)

	// the button
	btnW, btnH := float32(w), float32(h)
	button := giu.Layout{
		giu.ImageButton(t).Size(btnW, btnH).OnClick(fn),
	}
	
	// the container; needs to be padded to be larger than the button
	id := fmt.Sprintf("%s%s", id, childIdSuffix)
	con := giu.Child(id).
		Border(false).
		Size(float32(btnW+padBothSides), float32(btnH+padBothSides)).
		Layout(button).
		Flags(giu.WindowFlagsNoDecoration)

	return con
```

Reply:
why not: 0c63ad1